pandas/core/arrays/string_.py

```diff
@@ -731,6 +733,22 @@ def _putmask(self, mask: npt.NDArray[np.bool_], value) -> None:
         # base class implementation that uses __setitem__
         ExtensionArray._putmask(self, mask, value)
 
+    def isin(self, values: ArrayLike) -> npt.NDArray[np.bool_]:
+        if not isinstance(values, BaseStringArray):
+            if not lib.is_string_array(np.asarray(values), skipna=True):
+                values = np.array(
+                    [val for val in values if isinstance(val, str) or isna(val)],
+                    dtype=object,
+                )
+                if not len(values):
+                    return np.zeros(self.shape, dtype=bool)
+
+            values = self._from_sequence(values, dtype=self.dtype)
```

Comment:
This ensures that we convert the input `values` to a StringArray first, to make use of the construction validation (coercing to one missing value sentinel). 
But, this is only done if we actually get strings passed, and if not, we filter out non-strings. This is also what is done in the `ArrowStringArray` implementation, and this is to ensure we accept something like `isin(["string", 10])` with mixed-type values (I do wonder if we should deprecate that in general, but that's for a later, separate discussion).